### ISSUE TITLE
Python3 error in docutils-0.12

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1531]
sphinx-build -b html -d _build/doctrees  -v -v -T -P . _build/htmlnnns

gives:

```
[autodoc] import BibLib
[autodoc] => <module 'BibLib' from '/home/jarausch/Bib_NDev/IGPM_Bib_Doc/IGPM_Bib/BibLib.py'>
[autodoc] getattr(_, 'reduce')
[autodoc] => <built-in function reduce>
Exception occurred while building, starting debugger:
Traceback (most recent call last):
  File "/usr/lib64/python3.3/site-packages/sphinx/cmdline.py", line 254, in main
    app.build(force_all, filenames)
  File "/usr/lib64/python3.3/site-packages/sphinx/application.py", line 212, in build
    self.builder.build_update()
  File "/usr/lib64/python3.3/site-packages/sphinx/builders/__init__.py", line 214, in build_update
    'out of date' &#37; len(to_build))
  File "/usr/lib64/python3.3/site-packages/sphinx/builders/__init__.py", line 234, in build
    purple, length):
  File "/usr/lib64/python3.3/site-packages/sphinx/builders/__init__.py", line 134, in status_iterator
    for item in iterable:
  File "/usr/lib64/python3.3/site-packages/sphinx/environment.py", line 477, in update_generator
    self.read_doc(docname, app=app)
  File "/usr/lib64/python3.3/site-packages/sphinx/environment.py", line 624, in read_doc
    pub.publish()
  File "/usr/lib64/python3.3/site-packages/docutils/core.py", line 218, in publish
    self.apply_transforms()
  File "/usr/lib64/python3.3/site-packages/docutils/core.py", line 199, in apply_transforms
    self.document.transformer.apply_transforms()
  File "/usr/lib64/python3.3/site-packages/docutils/transforms/__init__.py", line 171, in apply_transforms
    transform.apply(**kwargs)
  File "/usr/lib64/python3.3/site-packages/docutils/transforms/universal.py", line 40, in apply
    footer_nodes = self.generate_footer()
  File "/usr/lib64/python3.3/site-packages/docutils/transforms/universal.py", line 62, in generate_footer
    settings._source)
  File "/usr/lib64/python3.3/site-packages/docutils/utils/__init__.py", line 461, in relative_path
    source_parts = os.path.abspath(source or type(target)('dummy_file')
TypeError: string argument without an encoding
> /usr/lib64/python3.3/site-packages/docutils/utils/__init__.py(461)relative_path()
-> source_parts = os.path.abspath(source or type(target)('dummy_file')
(Pdb) p target
b'/home/jarausch/Bib_NDev/IGPM_Bib_Doc/BibLib.rst'
(Pdb) type(target)
<class 'bytes'>
(Pdb) type(source)
<class 'NoneType'>
(Pdb) os.path.abspath(bytes('dummy_file'))
*** TypeError: string argument without an encoding
```

Is this a Python error? I'm using Python-3.3.5 and docutils-0.12.
Strangely enough, this occurs on one of two machines only.

Thanks for a hint,
Helmut

The following patch seems to fix it:

--- docutils/utils/**init**.py.ORIG     2014-08-06 18:26:21.000000000 +0200
+++ docutils/utils/**init**.py  2014-08-07 18:42:03.743833443 +0200
@@ -458,6 +458,8 @@

```
 If there is no common prefix, return the absolute path to `target`.
 """
```
-    if type(target) == bytes :
-      target= target.decode()
   source_parts = os.path.abspath(source or type(target)('dummy_file')
                                 ).split(os.sep)
   target_parts = os.path.abspath(target).split(os.sep)

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1531
- Originally reported by: Helmut Jarausch
- Originally created at: 2014-08-07T16:14:28.436

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-08-11 13:06:22+00:00_

I think you are using `docutils.conf` as:

```
[general]
source_link = true
```

That trigger the sphinx potentiality bug relates to Unicode/Bytes handling.
I'll try to fix it.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-08-11 15:12:00+00:00_
- On Python3 environment, docutils.conf with 'source_link=true' in the general section cause type error. Closes #1531

→ <<cset e984e381e8d670bd1be95877852649ccb21f2dcb>>
